extensions/wikia/JWPlayerTag/JWPlayerTagController.class.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+class JWPlayerTagController extends WikiaController {
+
+	const PARSER_TAG_NAME = 'jwplayer';
+	const DATA_MEDIA_ID_ATTR = 'data-media-id';
+	const ELEMENT_ID_PREFIX = 'jwPlayerTag';
+	const WIDTH_ATTR = 'height';
+	const STYLE_ATTR = 'height';
+
+	private $wikiaTagBuilderHelper;
+
+	public function __construct() {
+		parent::__construct();
+
+		$this->wikiaTagBuilderHelper = new WikiaTagBuilderHelper();
+	}
+
+	public static function onParserFirstCallInit( Parser $parser ): bool {
+		$parser->setHook( self::PARSER_TAG_NAME, [ new static(), 'renderTag' ] );
+
+		return true;
+	}
+
+	public function renderTag( $input, array $args, Parser $parser, PPFrame $frame ): string {
+		if ( !$this->validateArgs( $args ) ) {
+			return '<strong class="error">' . wfMessage( 'jwplayer-tag-could-not-render' )->parse() . '</strong>';
+		}
+
+		$script = JSSnippets::addToStack( [
```

Comment:
We could consider using ResourceLoader module and add it via `$parser->getOutput()->addModules()`